### ISSUE TITLE
NVEncC AV1 encoder not accepting input for profile or level

### ISSUE THREAD

**Issue by AlienBigfootZ** [staxrip/staxrip#1534]
NVEncC AV1 encoder not accepting input for profile or level in mode QVBR or VBR in constant quality mode.
Staxrip 2.42.2 / NVEncC 7.66. This is not new to this version. These input have never worked for me. Nvidia drivers 561.09 and older defaulted to a a main profile at level 4.0 in 1080p, 5.0 in 2160p which worked OK for my purposes. The default behavior is now to encode at level 7.3. Please provide the ability to adjust profile/tier/level. I checked this behavior in other encoding apps, they also encode at level 7.3 when left to auto with the newer nvidia drivers.
[Error Video encoding (v2.42.2).txt](https://github.com/user-attachments/files/17740472/Error.Video.encoding.v2.42.2.txt)

**Comment by Dendraspis** [comment]
And which GPU do you use?

**Comment by Dendraspis** [comment]
> Please provide the ability to adjust profile/tier/level.

I don't get your request ... you can already adjust all of them.  🥴 

>  I checked this behavior in other encoding apps, they also encode at level 7.3 when left to auto with the newer nvidia drivers.

When you set all of those parameters to `Auto` you won't get any error messages.


Keep in mind that StaxRip is just a GUI for the (native) encoders. StaxRip is responsable for the parameters, that are set, but the encoder is responsable for the encode and the outcome.

So in that case the parameters were set correctly, at least according to the NVEncC docs. Nevertheless I have opened this issue for you:
https://github.com/rigaya/NVEnc/issues/645

**Comment by Dendraspis** [comment]
No need to close this issue. Let's wait for the solution, maybe there have to be some adjustments made on StaxRips side to finally solve the issue, too.

**Comment by Dendraspis** [comment]
You see, the `--level` parameter has to be removed for AV1.  😅

**Comment by AlienBigfootZ** [comment]
I don't understand why "--level" would need to be omitted. It works with the nvenc implementation in another common video encoding application. Unfortunately that application does not support 2 pass encoding though.

**Comment by Dendraspis** [comment]
Did you check the outcome of those encodes?

**Comment by AlienBigfootZ** [comment]
Yes. The other encoder (when the level set to auto) yields a 7.3 with video drivers newer than 561.09. Setting the level to 4.0 in that other encoder results in a 4.0 encode on all versions of the nvidia driver that I tried. file size and bitrate are quite close to that from NVEncC.

**Comment by AlienBigfootZ** [comment]
Thanks anyway. Perhaps I'll check back later. I found out that the other encoder actually does do 2 pass av1.

**Comment by Dendraspis** [comment]
Would be interesting to find out what they do with the levels when they are not supported by the NVidia drivers.   😅 

Sure, and thanks for pointing that out.  👍 

Nevertheless let's keep it open as it is not solved yet...

**Comment by AlienBigfootZ** [comment]
As far as I can see, the Nvidia drivers have nothing to do with NVEncC not being able to set level. The only issue i have with the drivers is they don't default to the level they used to when level is set to auto. (4 for 1080p, 5 for 2160p) If NVEncC was able to set level the issue would be fixed. The drivers only pointed out the limitation in NVEncC.

**Comment by Dendraspis** [comment]
I don't think so...
NVEncC uses NVEnc which calls the API and when NVEncC is passing those parameters to the driver, it seems to say: ***I don't know/want "--level"***.

**Comment by AlienBigfootZ** [comment]
NVEncC v7.75 acts in the same fashion when leaving level in auto (encodes at level 7.3)
Level now has an effect as it changes the maximum bit rate. Tried in auto, level 4, land level 4.1. Level 4, and level 4.1 reduce the bitrate to a level appropriate to that level but still encodes at 7.3
Test 1 (auto) at 32.3 Mb/s, test 2 at level 4 at 11.6Mb/s and test 3 at level 4.1 at 19.2Mb/s. 
[2024-11-22 - 20.30.19 - Test level 4.1_staxrip.log](https://github.com/user-attachments/files/17878630/2024-11-22.-.20.30.19.-.Test.level.4.1_staxrip.log)
[2024-11-22 - 20.27.35 - Test level 4_staxrip.log](https://github.com/user-attachments/files/17878631/2024-11-22.-.20.27.35.-.Test.level.4_staxrip.log)
[2024-11-22 - 20.25.04 - Test_staxrip.log](https://github.com/user-attachments/files/17878632/2024-11-22.-.20.25.04.-.Test_staxrip.log)

**Comment by Dendraspis** [comment]
Yeah, that's what it is supposed to do .. now.
With a FullHD test file and an old NVidia driver it sets it to Level 4.0:
```
Format                         : AV1
Format/Info                    : AOMedia Video 1
Format profile                 : Main@L4.0
Codec ID                       : V_AV1
Duration                       : 3 min 33 s
Bit rate                       : 2 464 kb/s
Width                          : 1 920 pixels
Height                         : 1 080 pixels
```

Nevertheless nothing I can do here. StaxRip only passes the parameters to the encoder and it's their business what they do with those settings.
So if you wanna complain about the behavior and/or find out why level 7.3 is chosen, you have to head over. But I'm pretty sure the level is set by the driver and that it would be sort of impossible to find out why this happens. Would also mean to contact NVidia I guess. 
Maybe they will change/fix it soon, but who knows...

**Comment by Dendraspis** [comment]
Parameters will be removed in `v2.44.0`...
